course3/homework-unique_ptr/unique_ptr.cc

```diff
@@ -0,0 +1,111 @@
+#include "unique_ptr.h"
+
+#include <utility>
+
+
+template<class T>
+karun::unique_ptr<T>::unique_ptr() : mPtr(nullptr) {}
```

Comment:
I did that in my first implementation. However, if I use default, I also need to initialize mPtr to nullptr in the header file, else it contains a junk pointer. Since most other constructors overwrite the initial value in mPtr anyway, I decided to eliminate the initialization in the header file and just ensure all constructors initialize mPtr appropriately. This is probably a completely unnecessary optimization - is there actually any runtime cost to initializing a member in the header file and then overwriting it in a constructor?